pom.xml

```diff
@@ -344,7 +345,7 @@
                                 <goals>
                                     <goal>test</goal>
                                 </goals>
-                                <phase>test</phase>
+                                <phase>integration-test</phase>
```

Comment:
what does this do compared to `test` ?

Reply:
It runs in a different time in the maven lifecycle.
`test` runs after the code is compiled, and uses `target/classes` on the classpath to resolve the code.
`integration-test` runs after the jar is packaged, and uses the jar itself on the classpath instead of the classfiles.

The jar is a multi-release jar, and the module-info is contained in `META-INF/versions/9` which is automatically used if a jar is loaded. This doesn't work in a non-packaged layout like `target/classes`. So in order to pick up the module-info, the tests need to run on the jars. The module-info is required because the native-image builder always runs on the module system.